addons/udes_priorities/models/priorities.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api, _
+
+
+class UdesPriority(models.Model):
+    _name = "udes_priorities.priority"
+    _order = "sequence,id"
+
+    name = fields.Char(required=True)
+    reference = fields.Char(help="Unique identifier for the priority", required=True)
+
+    def _default_sequence_id(self):
+        return self.env.ref("udes_priorities.sequence_default")
+
+    sequence = fields.Integer(default=_default_sequence_id, help="Used to order the priorities")
+    description = fields.Text(help="A description about the purpose of the priority")
+    picking_type_ids = fields.Many2many(
+        "stock.picking.type",
+        string="Stock Transfer Types",
+        help="The stock picking types this priority is visible on. No picking types means all picking types",
+    )
+    group_ids = fields.Many2many(
+        "udes_priorities.priority_group", column1="priority_id", column2="group_id"
+    )
+
+    active = fields.Boolean(default=True)
+
+    _sql_constraints = [
+        ("priority_reference_unique", "unique(reference)", "reference must be unique")
+    ]
+
+    @api.multi
+    def _number_of_oustanding_pickings(self):
+        Picking = self.env["stock.picking"]
+        self.ensure_one()
+        return Picking.search_count(
+            [("priority", "=", self.reference), ("state", "not in", ("cancel", "done")),]
+        )
+
+    @api.multi
+    def _check_for_oustanding_pickings(self):
+        priorities_with_outstanding_picks = self.browse()
+        for priority in self:
+            if priority._number_of_oustanding_pickings() > 0:
+                priorities_with_outstanding_picks |= priority
+
+        if priorities_with_outstanding_picks:
+            raise ValueError(
+                _("There are outstanding pickings for priorities: {}").format(
+                    ", ".join(priorities_with_outstanding_picks.mapped("name"))
+                )
+            )
+
+    @api.multi
+    def write(self, values):
+        self._check_for_oustanding_pickings()
```

Comment:
This will prevent to update any field of the priority, are we sure we want this for fields like `description`?

Reply:
Probably we need a black lists of fields that cannot be changed when there are outstanding pickings.

Reply:
Done